### PR TITLE
Fields definition API

### PR DESCRIPTION
**Pull request by jsmadis**
**What this PR does / why we need it**:

This PR adds fields definition to the admin API.

**Which issue(s) this PR fixes** 

https://issues.redhat.com/browse/THREESCALE-4082

### CODE REVIEW

**Comment by jsmadis** [comment]
FYI this PR exposes an old bug in the fields definition (https://issues.redhat.com/browse/THREESCALE-7189). The bug will be fixed in the separate PR

File: config/routes.rb [hunk 1]
```diff
@@ -646,6 +646,8 @@ def call(env)
 
       resources :features, :controller => 'account_features', :except => [:new, :edit]
 
+      resources :fields_definitions, :controller => 'fields_definitions', :except => [:new, :edit]
```
Comment (josemigallas) [inline config/routes.rb:649]:
```suggestion
      resources :fields_definitions, controller: 'fields_definitions', except: %i[new edit]
```

File: config/webpacker.yml [hunk 2]
```diff
@@ -33,6 +33,7 @@ development:
   <<: *default
   compile: true
   webpack_compile_output: true
+  check_yarn_integrity: false
```
Comment (josemigallas) [inline config/webpacker.yml:36]: 🚨 
```suggestion
```

File: spec/acceptance/api/fields_definition_spec.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
```
Comment (josemigallas) [inline spec/acceptance/api/fields_definition_spec.rb:1]:
```suggestion
#frozen_string_literal: true

require 'rails_helper'
```

File: spec/acceptance/api/fields_definition_spec.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+resource 'FieldsDefinition' do
+  let(:resource) { FactoryBot.create(:fields_definition,
+                                     account: provider,
+                                     target: 'Account',
+                                     name: 'stuff',
+                                     choices: %w[Orange Apple Banana]) }
```
Comment (josemigallas) [inline spec/acceptance/api/fields_definition_spec.rb:8]: Unimportant, but I think this way is more consistent with the ~rest~ majority of the codebase:
```suggestion
  let(:resource) { FactoryBot.create(:fields_definition, account: provider,
                                                         target: 'Account',
                                                         name: 'stuff',
                                                         choices: %w[Orange Apple Banana]) }
```

**Review by josemigallas — CHANGES_REQUESTED** [review]
The requested change is because of the check integrity settings in the webpack config

**Review by josemigallas — COMMENTED** [review]
Looking at the spec file, should we also add cases for `show` and `destroy`?

**Review by hallelujah — APPROVED** [review]
:rocket:

**Comment by jsmadis** [comment]
I will add some tests, @josemigallas  is right :)

**Comment by jsmadis** [comment]
Rebased with the actual master because it contains the fix for the failing cucumber tests

**Review by josemigallas — APPROVED** [review]

**Pull request merged**